src/python/pants/backend/python/target_types_test.py

```diff
@@ -339,35 +341,77 @@ def test_parse_requirements_file() -> None:
     }
 
 
-def test_python_distribution_dependency_injection() -> None:
+def test_inject_python_distribution_dependencies() -> None:
     rule_runner = RuleRunner(
         rules=[
             inject_python_distribution_dependencies,
+            resolve_pex_entry_point,
+            resolve_python_distribution_entry_points,
+            *import_rules(),
+            *python_sources.rules(),
             QueryRule(InjectedDependencies, [InjectPythonDistributionDependencies]),
         ],
-        target_types=[PythonDistribution, PexBinary],
+        target_types=[PythonDistribution, PythonRequirementLibrary, PythonLibrary, PexBinary],
         objects={"setup_py": PythonArtifact},
     )
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            python_requirement_library(
+                name='ansicolors',
+                requirements=['ansicolors'],
+                module_mapping={'ansicolors': ['colors']},
+            )
+            """
+        ),
+    )
+    rule_runner.create_file("project/app.py")
     rule_runner.add_to_build_file(
         "project",
         dedent(
             """\
-            pex_binary(name="my_binary")
+            pex_binary(name="my_binary", entry_point="who_knows.module")
             python_distribution(
-                name="dist",
+                name="dist-a",
                 provides=setup_py(
-                    name='my-dist'
+                    name='my-dist-a'
                 ).with_binaries({"my_cmd": ":my_binary"})
             )
+
+            python_library(name="my_library", sources=["app.py"])
+            python_distribution(
+                name="dist-b",
+                provides=setup_py(
+                    name="my-dist-b"
+                ),
+                entry_points={
+                    "console_scripts":{
+                        "b_cmd": "project.app:main"
+                    }
```

Comment:
You'll also want to check using a `pex_binary` address, can even be the same one used above for `with_binaries`.

Reply:
Also a dep on `ansicolors`

Reply:
not sure what you me to test wrgt `ansicolors`.. ? (how/where?)
I see there is a requirement library for ansicolors in the test, that doesn't seem to do anything.. but don't know how to pull that in, in a useful way.